From: olh@suse.de
Subject: These macros are gone from glibc 2.4 includes.

arch/um/os-Linux/sys-i386/registers.c:134: error: 'JB_PC' undeclared (first use in this function)
arch/um/os-Linux/sys-i386/registers.c:134: error: (Each undeclared identifier is reported only once
arch/um/os-Linux/sys-i386/registers.c:134: error: for each function it appears in.)
arch/um/os-Linux/sys-i386/registers.c:135: error: 'JB_SP' undeclared (first use in this function)
arch/um/os-Linux/sys-i386/registers.c:136: error: 'JB_BP' undeclared (first use in this function)

Signed-off-by: Olaf Hering <olh@suse.de>

 arch/um/os-Linux/sys-i386/registers.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletion(-)

diff -r 19e39e4360ce arch/um/os-Linux/sys-i386/registers.c
--- a/arch/um/os-Linux/sys-i386/registers.c	Mon Mar  6 12:29:23 2006 -0500
+++ b/arch/um/os-Linux/sys-i386/registers.c	Mon Mar  6 12:31:30 2006 -0500
@@ -129,7 +129,10 @@ void get_safe_registers(unsigned long *r
 		memcpy(fp_regs, exec_fp_regs,
 		       HOST_FP_SIZE * sizeof(unsigned long));
 }
-
+/* from glibc 2.3 as found in sles9-i386 */
+#define JB_PC 5
+#define JB_SP 4
+#define JB_BP 3
 void get_thread_regs(union uml_pt_regs *uml_regs, void *buffer)
 {
 	struct __jmp_buf_tag *jmpbuf = buffer;
